test/test_util.py

```diff
@@ -3,19 +3,72 @@
 from datetime import datetime
 from time import sleep
 from typing import List
-from unittest.mock import Mock
+from unittest.mock import Mock, patch
 
 import copy
 import logging
 import myhoard.util as myhoard_util
 import os
+import pymysql
 import pytest
 import random
 import subprocess
 
 pytestmark = [pytest.mark.unittest, pytest.mark.all]
 
 
+@patch("pymysql.connect")
+@pytest.mark.parametrize(
+    "conn_options",
+    [
+        {
+            "password": "QR2$9Up2FAqF",
+            "port": 3306,
+            "user": "root",
+        },
+        {
+            "password": "QR2$9Up2FAqF",
+            "port": 3306,
+            "require_ssl": True,
+            "user": "root",
+        },
+        {
+            "ca_file": "ca-bundle.crt",
+            "db": "data",
+            "host": "localhost",
+            "password": "QR2$9Up2FAqF",
+            "port": 3306,
+            "timeout": 10.0,
+            "user": "admin",
+        },
+    ],
+)
+def test_mysql_cursor(connect_mock, conn_options):
+    with myhoard_util.mysql_cursor(**conn_options):
+        pass
+
+    timeout = conn_options.get("timeout", 4.0)
```

Comment:
Where does this `4.0` come from? It should probably be a constant in the code. If it's a magic const, that should be improved. Then you can import it in the test and use it here.

Reply:
I just used the same default from `util.py` (`DEFAULT_MYSQL_TIMEOUT = 4.0`).

Reply:
I personally do not like sharing constants between actual code and tests (for obvious reasons of making tests reliable, i.e. if we expect a timeout to be 10s, then our tests should enforce that, which means my tests should catch (fail because of) any change of my actual code that changes it to something else other than 10s). With that said, my approach is to either hard code something like this in my tests if it's only used once or extract it to tests-specific constants in case we need to reuse it.

Reply:
Okay, your explanation also makes sense and I'll resolve this thread! Thanks!